src/lava/magma/compiler/builders/interfaces.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+from abc import ABC, abstractmethod
+
+import typing as ty
+
+from lava.magma.compiler.channels.interfaces import AbstractCspPort
+if ty.TYPE_CHECKING:
+    from lava.magma.core.model.model import AbstractProcessModel
+
+
+class AbstractProcessBuilder(ABC):
```

Comment:
Let's get in the habit of writing class docstrings right away. 
Here we should state what Builders are and how this abstract class is supposed to be used.